File: frontend/lib/models/inquiry.dart
class Inquiry {
  final String id;
  final String? customerId;
  final String? serviceId;
  final String name;
  final String email;
  final String? phone;
  final String serviceType;
  final String message;
  final String status;
  final DateTime createdAt;

  Inquiry({
    required this.id,
    this.customerId,
    this.serviceId,
    required this.name,
    required this.email,
    this.phone,
    required this.serviceType,
    required this.message,
    required this.status,
    required this.createdAt,
  });

  factory Inquiry.fromJson(Map<String, dynamic> json) {
    return Inquiry(
      id: json['_id'],
      customerId: json['customer_id'] is Map ? json['customer_id']['_id'] : json['customer_id'],
      serviceId: json['service_id'] is Map ? json['service_id']['_id'] : json['service_id'],
      name: json['name'],
      email: json['email'],
      phone: json['phone'],
      serviceType: json['service_type'],
      message: json['message'],
      status: json['status'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  String get statusLabel {
    switch (status) {
      case 'new':
        return 'New';
      case 'in_discussion':
        return 'In Discussion';
      case 'completed':
        return 'Completed';
      default:
        return status;
    }
  }
}

File: backend_zip/backend/models/Inquiry.js
const mongoose = require('mongoose');

const inquirySchema = new mongoose.Schema({
  customer_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  service_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Service' },
  name: { type: String, required: true, trim: true },
  email: { type: String, required: true, trim: true },
  phone: { type: String, trim: true },
  service_type: { type: String, enum: ['ai_dev', 'digital_marketing'], required: true },
  message: { type: String, required: true },
  attachments: [{
    filename: String,
    originalName: String,
    mimetype: String,
    size: Number,
    url: String
  }],
  status: { type: String, enum: ['new', 'in_discussion', 'completed'], default: 'new' },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Inquiry', inquirySchema);
